src/main/java/com/google/api/codegen/transformer/php/PhpGapicSurfaceTransformer.java

```diff
@@ -292,4 +471,36 @@ private void addApiImports(GapicInterfaceContext context) {
 
     return apiMethods;
   }
+
+  private String generateConfigOutputPath(GapicInterfaceContext context, String name) {
+    GapicInterfaceConfig interfaceConfig = context.getInterfaceConfig();
+    String outputPath =
+        pathMapper.getOutputPath(
+            context.getInterfaceModel().getFullName(), context.getProductConfig());
+    return outputPath.substring(0, outputPath.length() - 5)
+        + "/resources/"
+        + Name.upperCamel(interfaceConfig.getInterfaceModel().getSimpleName())
+            .join(name)
+            .toLowerUnderscore()
+        + ".php";
+  }
+
+  private HttpRule getHttpRule(Map<FieldDescriptor, Object> optionFields) {
+    for (FieldDescriptor fieldDescriptor : optionFields.keySet()) {
+      if (fieldDescriptor.getFullName().equals("google.api.http")) {
+        return (HttpRule) optionFields.get(fieldDescriptor);
+      }
+    }
+
+    throw new IllegalStateException("A HttpRule option must be defined.");
```

Comment:
Another possibility we could consider - if there is no HttpRule option defined, instead of throwing a generation error, we just don't support the REST transport for that method, and we throw a runtime error if someone tries to use it, the same way we do for streaming. Is this:
1. Sensible, and does it ever come up?
2. Worth the effort?

Reply:
It seems if an HttpRule isn't defined and the method is not streaming that it would indicate an issue in the protos that needs to be resolved. I like the idea of notifying us of that being a potential issue early, rather than down the line when a user may run into it.

Reply:
That makes sense to me.